File: tsc/src/merge-arrays.ts
/**
 Do not return anything, modify nums1 in-place instead.
 */
export function mergeArrays(
  nums1: number[],
  m: number,
  nums2: number[],
  n: number
): void {
  let mi = m - 1;
  let nj = n - 1;

  for (let i = 0; i < m + n; i++) {
    if (nj < 0) {
      nums1[m + n - 1 - i] = nums1[mi];
      mi--;
      continue;
    }

    if (mi < 0) {
      nums1[m + n - 1 - i] = nums2[nj];
      nj--;
      continue;
    }

    if (nums1[mi] > nums2[nj]) {
      nums1[m + n - 1 - i] = nums1[mi];
      mi--;
    } else {
      nums1[m + n - 1 - i] = nums2[nj];
      nj--;
    }
  }
}

File: go/merge-arrays.go
package solution


func mergeArrays(nums1 []int, m int, nums2 []int, n int)  {

  var combined = m + n
  var mi = m - 1
  var ni = n - 1

  for i := combined - 1; i >= 0; i-- {
    if mi < 0 {
      nums1[i] = nums2[ni]
      ni--
      continue;
    }

    if(ni < 0) {
      nums1[i] = nums1[mi]
      mi--
      continue;
    }

    if nums1[mi] < nums2[ni] {
      nums1[i] = nums2[ni]
      ni--
    } else {
      nums1[i] = nums1[mi]
      mi--
    }
  }
}
